app/views/categories/_form.html.erb

```diff
@@ -0,0 +1,23 @@
+<%= form_with(model: category) do |form| %>
+  <% if category.errors.any? %>
+    <div style="color: red">
+      <h2><%= pluralize(category.errors.count, "error") %> prohibited this post from being saved:</h2>
+
+      <ul>
+        <% category.errors.each do |error| %>
+          <li><%= error.full_message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
+
+  <div>
+    <%= form.text_field :name, placeholder: "Name" %>
+  </div>
+  <div>
+    <%= form.file_field :icon %>
+  </div>
+  <div class="category-submit">
+    <%= form.submit %>
```

Comment:
- [ ] Your app is not displaying validation messages. When clicking on submit without filling out the fields, no validation errors were displayed but rather could submit though no category was created. As requested by the previous reviewer, kindly improve the user's experience while using our app. We are targeting the end-user to have a great experience while using our app.